### PR TITLE
added catch2

### PR DESCRIPTION
**Pull request by fnc12**

### CODE REVIEW

**Comment by Farwaykorse** [comment]
> @Farwaykorse can you please help me in added include path on osx - travis cannot compile unit tests #334
> Thanks in advance

I believe the best method is to let CMake do the work of finding the include paths.
Modify `/tests/CMakeLists.txt`:
````
find_package(Catch2 REQUIRED)
target_link_libraries(unit_tests PRIVATE sqlite_orm sqlite3 Catch2::Catch2)
````
This should work for Linux, OSX and Windows, no need to set the path yourself, unless you manually add the Catch2 library into your project.

File: .travis.yml [hunk 1]
```diff
@@ -44,6 +44,7 @@ matrix:
       language: cpp
       env:
         - SQLITE_ORM_OMITS_CODECVT: ON
+        - CPPFLAGS: "-I/usr/local/include"
```
Comment (Farwaykorse) [inline .travis.yml]:
```suggestion

```

File: .travis.yml [hunk 2]
```diff
@@ -62,7 +63,7 @@ matrix:
             - ninja
           update: true
       env:
-        - CPPFLAGS: "-I/usr/local/opt/llvm/include"
+        - CPPFLAGS: "-I/usr/local/opt/llvm/include -I/usr/local/include"
```
Comment (Farwaykorse) [inline .travis.yml]:
```suggestion
        - CPPFLAGS: "-I/usr/local/opt/llvm/include"
```

File: .travis.yml [hunk 3]
```diff
@@ -82,6 +83,7 @@ matrix:
             - ninja
           update: true
       env:
+        - CPPFLAGS: "-I/usr/local/include"
```
Comment (Farwaykorse) [inline .travis.yml]:
```suggestion

```

**Comment by Farwaykorse** [comment]
You might want to cancel those `Update .travis.yml` and `Merge pull request #335 from Farwaykorse/patch-1` builds on AppVeyor, they are not going to pass anyway.
https://ci.appveyor.com/project/fnc12/sqlite-orm/history

**Comment by fnc12** [comment]
@Farwaykorse osx now works but windows resulted to the error `CMake Error: Could not create named generator Visual Studio 15 2017 Win32`

File: appveyor.yml [hunk 4]
```diff
@@ -16,20 +12,23 @@ configuration:
...
 init:
-  - echo %APPVEYOR_BUILD_WORKER_IMAGE% - %configuration% - %arch%
+  - echo %APPVEYOR_BUILD_WORKER_IMAGE% - %configuration% - %PLATFORM%
   - if "%APPVEYOR_BUILD_WORKER_IMAGE%"=="Visual Studio 2017" (set vs=Visual Studio 15 2017)
   - if "%APPVEYOR_BUILD_WORKER_IMAGE%"=="Visual Studio 2015" (set vs=Visual Studio 14 2015)
   - if "%APPVEYOR_BUILD_WORKER_IMAGE%"=="Visual Studio 2013" (set vs=Visual Studio 12 2013)
-  - if "%arch%"=="Win64" (set generator="%vs% Win64")
-  - if "%arch%"=="Win32" (set generator="%vs%")
+  - if "%PLATFORM%"=="x64" (set generator="%vs% %PLATFORM%")
+  - if "%PLATFORM%"=="x86" (set generator="%vs% Win32")
```
Comment (Farwaykorse) [inline appveyor.yml]:
```suggestion
  - if "%PLATFORM%"=="x86" (set generator="%vs%")
```
Excuse me, my mistake.

File: appveyor.yml [hunk 5]
```diff
@@ -16,20 +12,23 @@ configuration:
...
 
 init:
-  - echo %APPVEYOR_BUILD_WORKER_IMAGE% - %configuration% - %arch%
+  - echo %APPVEYOR_BUILD_WORKER_IMAGE% - %configuration% - %PLATFORM%
   - if "%APPVEYOR_BUILD_WORKER_IMAGE%"=="Visual Studio 2017" (set vs=Visual Studio 15 2017)
   - if "%APPVEYOR_BUILD_WORKER_IMAGE%"=="Visual Studio 2015" (set vs=Visual Studio 14 2015)
   - if "%APPVEYOR_BUILD_WORKER_IMAGE%"=="Visual Studio 2013" (set vs=Visual Studio 12 2013)
-  - if "%arch%"=="Win64" (set generator="%vs% Win64")
-  - if "%arch%"=="Win32" (set generator="%vs%")
+  - if "%PLATFORM%"=="x64" (set generator="%vs% %PLATFORM%")
```
[the remark this answers is not in the document]
Comment (Farwaykorse) [reply appveyor.yml]: I was referencing a project using a different format to call the generator, which uses the `-A x64` and `-A Win32` flags.

[the remark this answers is not in the document]
Comment (Farwaykorse) [reply appveyor.yml]: Ignore this for now, this is an outdated method.
https://cmake.org/cmake/help/v3.13/generator/Visual%20Studio%2015%202017.html

**Comment by Farwaykorse** [comment]
That is my mistake, I'll commit a fix.

It seems this command uses an outdated format, that is incompatible CMake v3.14+ and with VS2019 :
[CMake v3.14 - VS2017 Platform Selection](https://cmake.org/cmake/help/v3.14/generator/Visual%20Studio%2015%202017.html#platform-selection)
[CMake v3.14 - VS2019 Platform Selection](https://cmake.org/cmake/help/v3.14/generator/Visual%20Studio%2016%202019.html#platform-selection)

**Comment by fnc12** [comment]
@Farwaykorse thank you for your help. I think it is time to add a list of people who helps me with the lib

**Pull request merged**